src/aurelia-binding.d.ts

```diff
@@ -184,10 +184,13 @@ export declare interface ICollectionObserverSplice<T = any> {
   value: T;
                                                    
   /**
-   * The type of change that has taken place. Valid options are "add", "delete", and "update".
-   * "update" is invalid for Set.
+   * The type of change that has taken place. Valid options are "splice", "add", "delete", "update", and "clear".
+   * 
+   * *Note:* "splice" is only valid for Array; "add", "update", and "clear" are invalid for Array.
+   * 
+   * *Note:* "update" is invalid for Set.
    */
-  type: "add" | "delete" | "update";
+  type: "splice" | "add" | "delete" | "update" | "clear";
```

Comment:
What's not so obvious from the code is that "splice" is actually normalized into "add"s and "delete"s, and "clear" is normalized into "delete"s. The original documentation is correct here for the end-users perspective. You will never receive "splice" or "clear" operations from this code.

Reply:
Interesting! That is good to know, I will revert this change back to the original.

Reply:
I actually just checked this for maps and sets and am receiving a type of clear when the clear method is called. Let me see if I can get a Gist going with this.

Reply:
Here's my Gist showing this: https://gist.run/?id=43a9adb8ba9004936e710bfef7269af4

Reply:
Huh. I stand corrected! And splice?

Reply:
No splice is correct I think https://gist.run/?id=afc8391c38d90dd5a398ed40c1698790

Reply:
In the gist, it shows that 'clear' is a valid type for Map | Set

Reply:
Awesome! Works for me, I'll re-add clear and make a note. Thanks guys!